### DISCUSSION TITLE
LED controller freezed when 'scheme is set to 2 or 3 or 4'

### DISCUSSION THREAD

**Discussion by johnnykang** [arendst/Tasmota#15195] [Q&A] [answered]
Hi,

I have an issue which I try to troubleshoot but I can't get my head around getting the log information as to why the device is freeze.

I have a RGB LED controller , flashed with Tasmota 11. And whenever I set the scheme to 2, 3 or 4, the device will freeze up after 5 mins or so. And I can't connect to the device as the webapi / wifi / connections are disconnected. The light will stop blinking / fading, it just shows the static colour. The only way to bring back the device is to disconnect the power and reconnect it. 

My question is, is there any way I can retrieve the previous device log (before the power disconnection?) I tried weblog 4 but it seems to erase itself after the power is disconnected. 

I know it is a bit vague as to i haven't provided the template or anything related to the hardware, but I am purely just looking for a way to get the crash log, if possible?

Idea?

** update
i am attaching the templates and relevant info.

```
21:38:18.589 CMD: Backlog Template; Module; GPIO 255
21:38:18.613 MQT: stat/tasmota_EF0065/RESULT = {"NAME":"LS5050C-TAS","GPIO":[32,0,0,0,416,0,0,0,417,0,418,0,0,0],"FLAG":0,"BASE":18}
21:38:18.836 MQT: stat/tasmota_EF0065/RESULT = {"Module":{"0":"LS5050C-TAS"}}
21:38:19.043 MQT: stat/tasmota_EF0065/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"416":"PWM1"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"417":"PWM2"},"GPIO13":{"0":"None"},"GPIO14":{"418":"PWM3"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
```

```21:40:19.756 CMD: Status 0
21:40:19.764 MQT: stat/tasmota_EF0065/STATUS = {"Status":{"Module":0,"DeviceName":"LED strip controller 2","FriendlyName":["LED strip controller 2"],"Topic":"tasmota_EF0065","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
21:40:19.772 MQT: stat/tasmota_EF0065/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"https://github.com/arendst/Tasmota/releases/download/v10.1.0/tasmota.bin.gz","RestartReason":"External System","Uptime":"0T11:22:49","StartupUTC":"2022-03-21T09:17:30","Sleep":50,"CfgHolder":4617,"BootCount":6,"BCResetTime":"2022-03-18T22:15:56","SaveCount":239,"SaveAddress":"F5000"}}
21:40:19.780 MQT: stat/tasmota_EF0065/STATUS2 = {"StatusFWR":{"Version":"11.0.0(tasmota)","BuildDateTime":"2022-02-12T14:13:50","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"394/699"}}
21:40:19.787 MQT: stat/tasmota_EF0065/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["homeip-iot",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A190000000000","00000080","00006200","00004000"]}}
21:40:19.802 MQT: stat/tasmota_EF0065/STATUS4 = {"StatusMEM":{"ProgramSize":620,"Free":380,"Heap":23,"ProgramFlashSize":1024,"FlashSize":2048,"FlashChipId":"1540C8","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8FDAC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","00000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6"}}
21:40:19.811 MQT: stat/tasmota_EF0065/STATUS5 = {"StatusNET":{"Hostname":"tasmota-EF0065-0101","IPAddress":"10.0.2.96","Gateway":"10.0.2.1","Subnetmask":"255.255.255.0","DNSServer1":"10.0.0.204","DNSServer2":"0.0.0.0","Mac":"10:52:1C:EF:00:65","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
21:40:19.818 MQT: stat/tasmota_EF0065/STATUS6 = {"StatusMQT":{"MqttHost":"10.0.2.86","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_EF0065","MqttUser":"mqtt","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
21:40:19.827 MQT: stat/tasmota_EF0065/STATUS7 = {"StatusTIM":{"UTC":"2022-03-21T20:40:19","Local":"2022-03-21T21:40:19","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"06:50","Sunset":"19:03"}}
21:40:19.834 MQT: stat/tasmota_EF0065/STATUS10 = {"StatusSNS":{"Time":"2022-03-21T21:40:19"}}
21:40:19.845 MQT: stat/tasmota_EF0065/STATUS11 = {"StatusSTS":{"Time":"2022-03-21T21:40:19","Uptime":"0T11:22:49","UptimeSec":40969,"Heap":23,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Dimmer":100,"Color":"00FFFF","HSBColor":"180,100,100","Channel":[0,100,100],"Scheme":0,"Fade":"ON","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"homeip-iot","BSSId":"24:5A:4C:63:CD:EC","Channel":1,"Mode":"11n","RSSI":90,"Signal":-55,"LinkCount":1,"Downtime":"0T00:00:07"}}}
```

**Comment by sfromis** [comment]
All log messages are gone when you have powered off. You could have the console running in browser tab, to at least have messages close to the point of freezing. You could also set up for Syslog or MqttLog for longer time logging. Best logs are if you can have a serial terminal connected as it happens.

**Reply by johnnykang** [reply]
thank you for your reply.

yup. i have tried setting weblog to 4 and have the browser displays the log. unfortunately, the log didn't reveal any cause as to why the device freeze right before the controller decided to go in hung.

serial terminal is not an option for me due to the work involved to get that going. :(

**Comment by Jason2866** [comment, marked as answer]
"Smells" like a power supply issue.

**Reply by johnnykang** [reply]
thanks for the reply.

I thought about the power supply as well, however, i have two identical controllers (with different power supply) that exhibit exact freeze behavior when setting scheme to either 2,3, or 4. And I have done the measure on the ampere used by the lights and the controller and it looks like the usage is within range. 

hence i have ruled out the power supply issue.

**Reply by sfromis** [reply]
One way to test if a power issue is likely is to run the same at low brightness. Your status info says full blast, which increases the risk of weaknesses causing unstable power.

Measuring amperes is useless to "rule out" such issues, as "power" is much more complex than a brick able to deliver some amount.

**Reply by johnnykang** [reply]
sure. i will give your suggestion
 a go and will report my finding back. 

Thanks mate.

**Reply by johnnykang** [reply]
ok. here is what i found. 

i tried dimmer = 1, 25 , 50, 80 , and all those values freezed up the controller after sometimes (it is pretty random, sometimes it takes 5 mins, sometimes 3 mins before the controller is required a DC disconnect).

scheme 0 works all the time with dimmer 100 . 

scheme 2,3,4 are the one always have issues.

**Reply by sfromis** [reply]
As a test, I've had a bulb cycling RGB `Scheme 4; Speed 1; Fade 0` for about 3 hours now, without seeing issues. 

If your controller is not well-designed, with suitable capacitors, it is not too implausible that fluctuations in power draw could lead to ripples in the supply to the MCU, making it unreliable. Obviously, that's pure speculation from my side.

**Reply by johnnykang** [reply]
> Scheme 4; Speed 1; Fade 0

cool. let me try that. thanks mate.

**Reply by johnnykang** [reply]
unfortunately, it exhibited the same freezing behavior with `Scheme 4; Speed 1; Fade 0` for me. :( 

> If your controller is not well-designed, with suitable capacitors, it is not too implausible that fluctuations in power draw could lead to ripples in the supply to the MCU, making it unreliable. Obviously, that's pure speculation from my side.

Yeah. I have 3 devices - 2 controllers and one [Mirabella LED portable lamp](https://templates.blakadder.com/mirabella_genio_I002940.html) , they all  freezed up after sometimes with Tasmota firmware with scheme 2,3,4. Those devices were working fine with its stock firmware when using the fading color effects.

i understand there are variables that can affect the reliability hence I was hoping if the log is able to give some hints. 

thanks for helping out. @sfromis .
